File: DHealth/app/src/main/java/com/example/hamid/dhealth/ui/Adapter/DoctorPatientListAdapter.java
package com.example.hamid.dhealth.ui.Adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.hamid.dhealth.R;
import com.example.hamid.dhealth.data.Preference.PreferenceKeys;
import com.example.hamid.dhealth.data.localdb.Entity.Doctor;
import com.example.hamid.dhealth.data.localdb.Entity.Patient;
import com.example.hamid.dhealth.ui.Activities.DoctorPatientDescriptionActivity;
import com.example.hamid.dhealth.utils.Utils;

import java.util.List;

public class DoctorPatientListAdapter extends RecyclerView.Adapter<DoctorPatientListAdapter.ViewHolder> {

    private Context mContext;
    private List<Doctor> doctorList;
    private List<Patient> patientList;
    private String profile_type;


    public DoctorPatientListAdapter(Context context, List<Doctor> doctorList, List<Patient> patientList, String profile_type) {
        this.mContext = context;
        this.doctorList = doctorList;
        this.patientList = patientList;
        this.profile_type = profile_type;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.doctor_patient_card, parent, false);

        DoctorPatientListAdapter.ViewHolder viewHolder = new ViewHolder(itemView, new DoctorPatientListAdapter.ViewHolder.IItemClickListener() {

            @Override
            public void onItemClick(View caller, int position) {

                Intent intent = new Intent(mContext, DoctorPatientDescriptionActivity.class);
                if (profile_type.equalsIgnoreCase(PreferenceKeys.LBL_DOCTOR)) {
                    Patient patient = patientList.get(position);
                    intent.putExtra(DoctorPatientDescriptionActivity.DATA, patient);
                } else {
                    Doctor doctor = doctorList.get(position);
                    intent.putExtra(DoctorPatientDescriptionActivity.DATA, doctor);
                }
                mContext.startActivity(intent);
            }
        });

        return viewHolder;
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, int position) {
        if (profile_type.equalsIgnoreCase(PreferenceKeys.LBL_DOCTOR)) {
            Patient patient = patientList.get(position);
            holder.tv_name.setText(patient.getFirst_name() + " " + patient.getLast_name());
            holder.iv_dp.setImageBitmap(Utils.decodeBase64(patient.getDp()));
        } else {
            Doctor doctor = doctorList.get(position);
            holder.tv_name.setText(doctor.getFirst_name() + " " + doctor.getLast_name());
            holder.iv_dp.setImageBitmap(Utils.decodeBase64(doctor.getDp()));
        }


    }


    @Override
    public int getItemCount() {

        if (profile_type.equalsIgnoreCase(PreferenceKeys.LBL_DOCTOR)) {
            if (patientList != null)
                return patientList.size();
            else return 0;
        } else {
            if (doctorList != null)
                return doctorList.size();
            else return 0;
        }

    }


    public void setDoctorList(List<Doctor> doctorList) {
        this.doctorList = doctorList;
        notifyDataSetChanged();
    }

    public void setPatientList(List<Patient> patientList) {
        this.patientList = patientList;
        notifyDataSetChanged();
    }


    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        public TextView tv_name;
        public ImageView iv_display;
        public ImageView iv_dp;

        private ViewHolder.IItemClickListener itemClickListener;


        public ViewHolder(View view, IItemClickListener itemClickListener) {
            super(view);

            tv_name = (TextView) view.findViewById(R.id.tv_name);
            iv_display = (ImageView) view.findViewById(R.id.iv_dp);
            iv_dp = (ImageView) view.findViewById(R.id.iv_dp);

            this.itemClickListener = itemClickListener;
            view.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            itemClickListener.onItemClick(view, getLayoutPosition());
        }

        public interface IItemClickListener {
            public void onItemClick(View caller, int position);
        }

    }


}

File: DHealth-Android/Kotlin/app/src/main/java/com/example/hamid/dhealth/ui/Adapter/DoctorPatientListAdapter.kt
package com.example.hamid.dhealth.ui.Adapter

import android.content.Context
import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import com.example.hamid.dhealth.R
import com.example.hamid.dhealth.data.Preference.PreferenceKeys
import com.example.hamid.dhealth.data.localdb.Entity.Doctor
import com.example.hamid.dhealth.data.localdb.Entity.Patient
import com.example.hamid.dhealth.ui.Activities.DoctorPatientDescriptionActivity
import com.example.hamid.dhealth.utils.Utils

class DoctorPatientListAdapter(private val mContext: Context, private var doctorList: List<Doctor>?, private var patientList: List<Patient>?, private val profile_type: String) : RecyclerView.Adapter<DoctorPatientListAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val itemView = LayoutInflater.from(parent.context)
                .inflate(R.layout.doctor_patient_card, parent, false)

        return ViewHolder(itemView, object : ViewHolder.IItemClickListener {

            override fun onItemClick(caller: View, position: Int) {

                val intent = Intent(mContext, DoctorPatientDescriptionActivity::class.java)
                if (profile_type.equals(PreferenceKeys.LBL_DOCTOR, ignoreCase = true)) {
                    val patient = patientList!![position]
                    intent.putExtra(DoctorPatientDescriptionActivity.DATA, patient)
                } else {
                    val doctor = doctorList!![position]
                    intent.putExtra(DoctorPatientDescriptionActivity.DATA, doctor)
                }
                mContext.startActivity(intent)
            }
        })
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        if (profile_type.equals(PreferenceKeys.LBL_DOCTOR, ignoreCase = true)) {
            val patient = patientList!![position]
            holder.tv_name.text = patient.first_name + " " + patient.last_name
            if (patient.dp!!.length > 4)
                holder.iv_dp.setImageBitmap(Utils.decodeBase64(patient.dp!!))
        } else {
            val doctor = doctorList!![position]
            holder.tv_name.text = doctor.first_name + " " + doctor.last_name
            if (doctor.dp!!.length > 4)
                holder.iv_dp.setImageBitmap(Utils.decodeBase64(doctor.dp!!))
        }


    }


    override fun getItemCount(): Int {

        return if (profile_type.equals(PreferenceKeys.LBL_DOCTOR, ignoreCase = true)) {
            if (patientList != null)
                patientList!!.size
            else
                0
        } else {
            if (doctorList != null)
                doctorList!!.size
            else
                0
        }

    }


    fun setDoctorList(doctorList: List<Doctor>) {
        this.doctorList = doctorList
        notifyDataSetChanged()
    }

    fun setPatientList(patientList: List<Patient>) {
        this.patientList = patientList
        notifyDataSetChanged()
    }


    class ViewHolder(view: View, itemClickListener: IItemClickListener) : RecyclerView.ViewHolder(view), View.OnClickListener {

        var tv_name: TextView
        var iv_display: ImageView
        var iv_dp: ImageView

        private val itemClickListener: ViewHolder.IItemClickListener


        init {

            tv_name = view.findViewById(R.id.tv_name)
            iv_display = view.findViewById(R.id.iv_dp)
            iv_dp = view.findViewById(R.id.iv_dp)

            this.itemClickListener = itemClickListener
            view.setOnClickListener(this)
        }

        override fun onClick(view: View) {
            itemClickListener.onItemClick(view, layoutPosition)
        }

        interface IItemClickListener {
            fun onItemClick(caller: View, position: Int)
        }

    }


}
